test/canary-environment/tests/mz_compute_frontiers.sql

```diff
@@ -0,0 +1,12 @@
+-- Copyright Materialize, Inc. and contributors. All rights reserved.
+--
+-- Use of this software is governed by the Business Source License
+-- included in the LICENSE file at the root of this repository.
+--
+-- As of the Change Date specified in that file, in accordance with
+-- the Business Source License, use of this software will be governed
+-- by the Apache License, Version 2.0.
+
+select *
+from mz_internal.mz_compute_frontiers
+where to_timestamp(time::text::bigint/1000) < to_timestamp(mz_now()::text::bigint/1000) - INTERVAL '1 minute'
```

Comment:
What is the purpose here? Why don't we compare the outcome with the expected one?

Reply:
This is a dbt-style test (as I am dogfooding dbt for this task). A dbt-style test is a SELECT statement that should return no rows. If it returns any rows, the `dbt test` command, which runs all those statements, will report a failure.

This particular query makes sure that all the dataflows are advancing and there is no dataflow whose last advance is more than 1 minute into the past. The extra casts and stuff is because the `time` column is of type `mz_timestamp`, which is not castable or comparable in almost any fashion unless converted to a normal timestamp using chained casts and divisions.